### PR TITLE
Remove eagerly created tasks

### PR DESCRIPTION
**Pull request by blindpirate**

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test this

File: build-logic/jvm/src/main/kotlin/gradlebuild.api-parameter-names-index.gradle.kts [hunk 1]
```diff
@@ -32,20 +32,12 @@ val parameterNamesIndex by tasks.registering(ParameterNamesIndex::class) {
     )
     classpath.from(main.compileClasspath)
     classpath.from(tasks.named<JavaCompile>("compileJava"))
+    classpath.from(tasks.named<GroovyCompile>("compileGroovy"))
     destinationFile.set(
         project.layout.buildDirectory.file(moduleIdentity.baseName.map { "generated-resources/$it-parameter-names/$it-parameter-names.properties" })
     )
 }
 
-tasks.withType<GroovyCompile>() {
```
Comment (blindpirate) [inline build-logic/jvm/src/main/kotlin/gradlebuild.api-parameter-names-index.gradle.kts:40]: Not sure why it was like this, but removing it seems not break things.

**Review by wolfs — APPROVED** [review]
LGTM!

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Pull request merged**